RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/service/TechnicianService.java

```diff
@@ -277,7 +278,66 @@ public static TechnicianDto technicianToDTO(Technician tech) {
 		
 	}
 	
+	@Transactional // TODO check if this is what is needed for this method: this method supposed to remove absolutely all the timeslots and appointments from technicians shedule
+	public String deleteFullTechnicianWorkSchedule(String email) throws Exception{
+		
+		if(email == null) {
+			throw new Exception("Email cannot be empty.");
+		}
+
+		Technician technician = technicianRepository.findTechnicianByEmail(email);
+		
+		if(technician == null) {
+			throw new Exception("Technician not found.");
+		}
+		
+		// removing all the appointments
+		for (Appointment appointmentToRemove: technician.getAppointments()) {
+			technician.getAppointments().remove(appointmentToRemove);
+		}
+		// removing all the timeslots
+		for (TimeSlot timeSlotToRemove: technician.getTimeslots()) {
+			technician.getTimeslots().remove(timeSlotToRemove);
+		}
+
+		technicianRepository.save(technician);
+		
+		return "All work hour timeslots and associated appointments for technician " + email + " were successfully removed.";
+	}
 	
+	@Transactional // TODO check if this is what is needed for this method: method supposed to remove the specific timeSlot and appointment from technician schedule
+	public String deleteSpecificTechnicianTimeSlotAndAppointmentsFromSchedule(String email, TimeSlot slotToDelete) throws Exception{
+		boolean removedAppointments = false;
+		boolean removedTimeSlot = false;
+		
+		if (email == null) {
+			throw new Exception("Email cannot be empty.");
+		}
+
+		Technician technician = technicianRepository.findTechnicianByEmail(email);
+		if (technician == null) {
+			throw new Exception("Technician not found.");
+		}
+		// First removing all the appointments within the chosen work time slot
+		for (Appointment appointmentToDelete: technician.getAppointments()) {
+			if (appointmentToDelete.getTimeSlot().equals(slotToDelete)) {
+				technician.getAppointments().remove(appointmentToDelete);
+				removedAppointments = true;
+			}
+		}
+		// Second removing the chosen timeslot itself
+		for (TimeSlotDto timeSlot: getWorkHours(email)) {
+			if (timeSlot.getStartDateTime().equals(slotToDelete.getStartDateTime()) 
+					&& timeSlot.getEndDateTime().equals(slotToDelete.getEndDateTime())) {
+				technician.getTimeslots().remove(slotToDelete);
```

Comment:
Here too, we need to add `timeSlotRepository.deleteById(timeSlotToRemove.getTimeSlotID())`?
```suggestion
			timeSlotRepository.deleteById(timeSlotToRemove.getTimeSlotID()); //delete it from TimeSlot Repository
				technician.getTimeslots().remove(slotToDelete);
```

Reply:
In the case that the timeSlotId is null this will throw an exception which will cause the whole thing to crash. So I think it's best to remove the object